File: LABA5/parserPack/FloatParser.java
package parserPack;

import exceptionsPack.NullFieldException;

public class FloatParser {
    public static Float parse(String line) throws NumberFormatException, NullFieldException {
        if (line.isEmpty()) throw new NullFieldException();
        line = line.replaceAll("_", "").replaceAll(",", ".");
        return Float.parseFloat(line);
    }
}

File: LABA5kt/src/parserPack/FloatParser.kt
package parserPack

import exceptionsPack.NullFieldException

object FloatParser {
    /**
     * Parses a string to a Float after replacing spaces, underscores, and commas with dots.
     *
     * @param  line  the string to be parsed
     * @return      the parsed Float value
     */
    @Suppress("NAME_SHADOWING")
    @Throws(NumberFormatException::class, NullFieldException::class)
    fun parse(line: String): Float {
        var line = line
        line = line.replace(" ".toRegex(), "").replace("_".toRegex(), "").replace(",".toRegex(), ".")
        if (line.isEmpty()) throw NullFieldException()
        return line.toFloat()
    }
}